packages/riipen-ui/src/components/MenuList.test.jsx

```diff
@@ -0,0 +1,299 @@
+      );
+
+      expect(selectChange).toHaveBeenLastCalledWith(0, undefined);
+    });
+
+    it("does not select first element on list open with false autoFocus", () => {
+      const autoFocus = false;
+      const selectChange = jest.fn();
+
+      mount(
+        <MenuList autoFocus={autoFocus} selectChange={selectChange}>
+          <MenuItem>one</MenuItem>
+          <MenuItem>two</MenuItem>
+        </MenuList>
+      );
+
+      expect(selectChange).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("children prop", () => {
+    it("displays given non null children", () => {
+      const child = <MenuItem>one</MenuItem>;
+
+      const wrapper = mount(
+        <MenuList>
+          {null}
+          {child}
+        </MenuList>
+      );
+
+      const childrenNodes = wrapper
+        .find("MenuList")
+        .childAt(0)
+        .children();
+      expect(childrenNodes.containsMatchingElement(child)).toEqual(true);
+      expect(childrenNodes).toHaveLength(1);
+    });
+  });
+
+  describe("classes prop", () => {
+    it("applies custom classes", () => {
+      const classes = ["classOne"];
+
+      const wrapper = mount(
+        <MenuList classes={classes}>
+          <MenuItem />
+        </MenuList>
+      );
+
+      expect(
+        wrapper
+          .find("MenuList")
+          .childAt(0)
+          .hasClass(classes[0])
+      ).toEqual(true);
+    });
+
+    it("appends higher order values to classes prop with withClass decorator", () => {
+      const classes = ["riipen", "riipen-menulist"];
+
+      const wrapper = mount(
+        <MenuList>
+          <MenuItem />
+        </MenuList>
+      );
+
+      expect(
+        wrapper
+          .find("MenuList")
+          .props()
+          .classes.sort()
+      ).toEqual(classes.sort());
+    });
+  });
+
+  describe("color prop", () => {
+    it("passes color prop into children nodes", () => {
+      const color = "secondary";
+
+      const wrapper = mount(
+        <MenuList color={color}>
+          <MenuItem>one</MenuItem>
+        </MenuList>
+      );
+
+      expect(wrapper.find("MenuItem").props().color).toEqual(color);
+    });
+
+    it("gives an error with invalid color", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(
+        <MenuList color="pink">
+          <MenuItem />
+        </MenuList>
+      );
+      expect(errors).toHaveBeenCalled();
+    });
+  });
+
+  describe("selectChange prop", () => {
+    it("invokes selectChange with a null idx when div is clicked", () => {
+      const selectChange = jest.fn();
+
+      const wrapper = mount(
+        <MenuList selectChange={selectChange}>
+          <MenuItem>one</MenuItem>
+        </MenuList>
+      );
+
+      wrapper
+        .find("MenuList")
+        .childAt(0)
+        .simulate("click");
+
+      expect(selectChange).toHaveBeenLastCalledWith(null, expect.any(Object));
+    });
+
+    describe("when variant is selection", () => {
```

Comment:
It looks like this describe block could be removed and the test could be moved to variant. The test name for this test and the test at line 262 sound very similar.

Reply:
Ya I am also confused about this describe block. Looking at the two tests within it, I wonder if they are better off being part of the children tests, or even remove the describe block entirely and just put those two tests at the same level of the other selectChange test. Up to you.

Reply:
I like them being part of children tests